embulk-output-redshift/src/main/java/org/embulk/output/redshift/RedshiftCopyBatchInsert.java

```diff
@@ -86,28 +93,37 @@ public void flush() throws IOException, SQLException
     {
         File file = closeCurrentFile();  // flush buffered data in writer
 
-        // TODO multi-threading
-        new UploadAndCopyTask(file, batchRows, s3KeyPrefix + UUID.randomUUID().toString()).call();
-        new DeleteFileFinalizer(file).close();
+        UploadAndCopyTask task = new UploadAndCopyTask(file, batchRows, s3KeyPrefix + UUID.randomUUID().toString());
+        executorService.submit(task);
 
         fileCount++;
         totalRows += batchRows;
         batchRows = 0;
 
         openNewFile();
-        file.delete();
     }
 
     @Override
     public void finish() throws IOException, SQLException
     {
         super.finish();
+
+        executorService.shutdown();
+        try {
+            executorService.awaitTermination(Long.MAX_VALUE, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+
         logger.info("Loaded {} files.", fileCount);
     }
 
     @Override
     public void close() throws IOException, SQLException
     {
+        if (!executorService.isShutdown()) {
+            executorService.shutdown();
```

Comment:
when `finish()` causes exception during `executorService.awaitTermination`, flow comes here. `ExecutorService#shutdown()` unlikely works. So, we should call `#shutdownNow()`.